File: WebNote/WebNote/Models/NoteRepository.cs
using System.Collections.Generic;
using System.Configuration;
using System.Linq;
using MongoDB.Bson;
using MongoDB.Driver;
using MongoDB.Driver.Builders;
using MongoDB.Driver.Linq;
using Builders = MongoDB.Driver.Builders;

namespace DeveloperMediaDemo.Models
{
    /// <summary>
    /// This is a simple demo repository with no real database connection
    /// </summary>
    public class NoteRepository
    {
        private const string ConnectionStringName = "MongoDB";
        private const string DatabaseName = "WebNote";
        private const string CollectionName = "Notes";

        private readonly MongoCollection<Note> notes;

        public NoteRepository()
        {
            string connectionString = ConfigurationManager.ConnectionStrings[ConnectionStringName].ConnectionString;
            MongoClient client = new MongoClient(connectionString);
            MongoServer server = client.GetServer();
            MongoDatabase database = server.GetDatabase(DatabaseName);
            notes = database.GetCollection<Note>(CollectionName);            
        }

        public void Create(Note item)
        {
            notes.Insert(item);
        }

        public Note Read(string id)
        {
            return (from n in notes.AsQueryable()
                    where n.Id == id
                    select n).First();
        }

        public IEnumerable<Note> ReadAll()
        {
            return (from n in notes.AsQueryable()
                    orderby n.Id ascending
                    select n).ToList();
        }        
        
        public void Update(Note item)
        {
            Note note = Read(item.Id);

            note.Title = item.Title;
            note.Message = item.Message;
            note.Categories = item.Categories;

            var query = Query.EQ("_id", ObjectId.Parse(item.Id));
            notes.Update(query, Builders.Update.Replace(note));
        }

        public void Delete(string id)
        {
            notes.Remove(Query.EQ("_id", ObjectId.Parse(id)));
        }
    }
}

File: WebNotePHP/module/Application/src/Application/Controller/NoteRepository.php
<?php

namespace Application\Controller;

class NoteRepository {

    const connectionString = "mongodb://localhost";

    private $notes;

    public function __construct() {

        $client = new \MongoClient();
        $database = $client->WebNote;
        $this->notes = $database->Notes;
    }

    public function create($item)
    {
        $this->notes->insert($item);
        return $item["_id"]->{'$id'};
    }

    public function read($id)
    {
        // The most common mistake is attempting to use a string to match a MongoId.
        $id = new \MongoId($id);
        $doc = $this->notes->findOne(array('_id' => $id));
        return $this->docToArray($doc);;
    }

    public function readAll()
    {
        $cursor = $this->notes->find();

        $result = [];
        foreach ($cursor as $doc) {
            $result[] = $this->docToArray($doc);
        }
        return $result;
    }

    public function update($item)
    {
        $filter = array('_id' => new \MongoId($item["Id"]));

        $update = array('$set' => array(
            'Title' => $item["Title"],
            'Message' => $item["Message"],
            'Categories' => $item["Categories"])
        );

        $this->notes->update($filter, $update);
    }

    public function delete($id)
    {
        $id = new \MongoId($id);
        $this->notes->remove(array('_id' => $id));
    }

    private function docToArray($doc) {

        return array(
            'Id' => $doc["_id"]->{'$id'},
            'Title' => $doc["Title"],
            'Message' => $doc["Message"],
            'Added' => date(DATE_ISO8601, $doc["Added"]->sec),
            'Categories' => $doc["Categories"]
        );
    }
}